legacy/include/distconv/dnn_backend/dace_backend.hpp

```diff
@@ -0,0 +1,271 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright 2019-2023 Lawrence Livermore National Security, LLC and other
+// DiHydrogen Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: Apache-2.0
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <map>
+#include <tuple>
+
+#include "dnn_backend.hpp"
+
+namespace distconv
+{
+/** @brief Type of JIT-compiled convolution **/
+enum ConvType
+{
+    FORWARD = 0,
+    BACKWARD_FILTER,
+    BACKWARD_DATA
+};
+
+/** @brief Convolution parameters **/
+struct ConvParams
+{
+    int pads[3];
+    int strides[3];
+    int dilation[3];
+    int groups;
+
+    bool operator<(const ConvParams& other) const;
+    bool operator==(const ConvParams& other) const;
```

Comment:
Make non-member non-friend binary operators? (Doesn't really matter since this isn't _really_ user-facing, but since you've chosen `struct` and it's all `public`, I'd prefer the "typical" interface for binary ops.)